### PR TITLE
Add bold support for CJK characters

### PR DESCRIPTION
**Pull request by wangyingfang**
Use system font family to draw characters of CJK, but don't support bold font on iOS. Add similar logic as Android side.
image
image

### CODE REVIEW

**Review by louwers — COMMENTED** [review]
Can't you use a bold font?

File: platform/darwin/src/local_glyph_rasterizer.mm [hunk 1]
```diff
@@ -266,6 +279,11 @@ CGContextHandle context(CGBitmapContextCreate(
     
     CTLineDraw(*line, *context);
     
+    // Release the bold font if it was created
+    if (boldFont) {
+        CFRelease(boldFont);
```
Comment (louwers) [inline platform/darwin/src/local_glyph_rasterizer.mm:284]: This is never called if the function throws. Should use `CFHandle`.

**Comment by louwers** [comment]
I wasn't aware this is exactly what MapLibre Android and MapLibre GL JS do...

File: platform/darwin/src/local_glyph_rasterizer.mm [hunk 2]
```diff
@@ -191,16 +191,29 @@ CFDictionaryRefHandle attributes(
...
 
+    // Can't use CTFontRefHandle
+    // because the boldFont will be released after it is out of the isBold condition
+    CTFontRef boldFont = NULL;
+    if (isBold)
+    {
+        // Create a bold variant of the font
+        boldFont = CTFontCreateCopyWithSymbolicTraits(font, 0.0, NULL, kCTFontBoldTrait, kCTFontBoldTrait);
+        if (!boldFont) {
+            throw std::runtime_error("Unable to create bold font");
```
Comment (louwers) [inline platform/darwin/src/local_glyph_rasterizer.mm:211]: Maybe instead of throwing you can log an error and set `boldFont` to `font`.

**Review by louwers — APPROVED** [review]
I added a test. Please take a look if it looks OK to you.

Also, please have a look at my comments. Thank you.

**Comment by random3940** [comment]
I find it peculiar that we need to figure out bold flag from the name, but change seems to reflect what Android does and fixes the bug. LTGM.

**Review by random3940 — APPROVED** [review]

File: platform/darwin/src/local_glyph_rasterizer.mm [hunk 3]
```diff
@@ -191,16 +193,26 @@ CFDictionaryRefHandle attributes(
...
     CFStringRefHandle string(CFStringCreateWithCharacters(NULL, reinterpret_cast<UniChar*>(&glyphID), 1));
     if (!string) {
         throw std::runtime_error("Unable to create string from codepoint");
     }
 
+    // Create a bold variant of the font
+    CTFontRefHandle boldFont(CTFontCreateCopyWithSymbolicTraits(font, 0.0, NULL, kCTFontBoldTrait, kCTFontBoldTrait));
+    if (!boldFont) {
+        CFStringRefHandle familyNameHandle(CTFontCopyFamilyName(font));
+        [MLNNativeNetworkManager.sharedManager errorLog:@"Unable to create bold font for %@", *familyNameHandle];
```
Comment (louwers) [inline platform/darwin/src/local_glyph_rasterizer.mm:209]: I think using `Log::Error` is more appropriate here. Otherwise LGTM.

**Pull request merged**